### PR TITLE
@xtina-starr => Partner articles show images on mobile

### PR DESCRIPTION
**Pull request by eessex**
Updates `/mobile/components/article` to use `image_collection` section over `image` or `artwork`

### CODE REVIEW

**Comment by eessex** [comment]
@xtina-starr the images were broken because the `/mobile/article` app was using outdated section types `image` and `artworks`, which were long-ago replaced with `image_collection` -- would be glad to talk through in person so you can get a better idea of the different places where articles are rendered. 

This app I mistakenly believed was deprecated already -- created this related issue, https://artsyproduct.atlassian.net/browse/GROW-731

File: src/mobile/components/article/templates/sections/slideshow.jade [hunk 1]
```diff
@@ -6,11 +6,9 @@ if section.items.length > 0
     - item = section.items[0]
     case item.type
       when 'image'
```
Comment (xtina-starr) [inline src/mobile/components/article/templates/sections/slideshow.jade]: I'm curious as to why this wouldn't need to change to `image_collection` as well like in the `/article/templates/index.jade` file?

File: src/mobile/components/article/templates/sections/images.jade [hunk 2]
```diff
@@ -0,0 +1,48 @@
+mixin printArtistName(artist)
```
Comment (xtina-starr) [inline src/mobile/components/article/templates/sections/images.jade]: This seems to be pretty similar to the old `article/templates/sections/artworks.jade` file. I'm not 100% clear of the reason behind the name change.

Comment (eessex) [reply src/mobile/components/article/templates/sections/images.jade]: Yeah barely changed any code, but moved the artwork rendering into a mixin. The difference is that the old files cycled only through artworks, but we need to instead map an array containing a mix of artworks and images.

File: src/mobile/components/article/templates/sections/slideshow.jade (hunk 1 above)
Comment (eessex) [reply src/mobile/components/article/templates/sections/slideshow.jade]: The slideshow is a section type, it is a sibling of of `image_collection` -- here we are rendering individual items in a slideshow, it mimics behavior of image_collection but with the added ability to render a video.

**Comment by xtina-starr** [comment]
Gotcha! I just had those question to better understand. Thanks for explaining. This looks good to me. Gonna merge now.

**Pull request merged**